### PR TITLE
(#2500) Add E2E tests for template parameters

### PR DESCRIPTION
**Pull request by AdmiringWorm**
This commit updates the end to end tests to add regression
testing for the incoming template parameters.

## Description Of Changes

This pull request is a followup from the PR #2507 that implemented the functionality to list parameters in templates.

## Motivation and Context

To have E2E regression tests available.

## Testing

1. Ran all template command tests in test kitchen

## Change Types Made

* [ ] Bug fix (non-breaking change)
* [ ] Feature / Enhancement (non-breaking change)
* [ ] Breaking change (fix or feature that could cause existing functionality to change)
* [ ] PowerShell code changes.
* [x] Tests

## Related Issue

#2500
https://app.clickup.com/t/20540031/ENGTASKS-1480

## Change Checklist

* [ ] Requires a change to the documentation
* [ ] Documentation has been updated
* [ ] Tests to cover my changes, have been added
* [x] All new and existing tests passed.
* [ ] PowerShell v2 compatibility checked.

### CODE REVIEW

File: tests/chocolatey-tests/choco-template.Tests.ps1 [hunk 1]
```diff
@@ -35,6 +35,10 @@ Describe "choco <_>" -ForEach @(
             $Output.Lines | Should -Contain 'msi 1.0.2'
             $Output.Lines | Should -Contain 'zip 1.0.0'
         }
+
+        It "Displays how many custom templates are available" {
+            $Output.Lines | Should -Contain "2 Custom templates found at $env:ChocolateyInstall\templates"
+        }
     }
 
     Context "Running with list subcommand" {
```
Comment (corbob) [inline tests/chocolatey-tests/choco-template.Tests.ps1:44]: This wasn't from your change, but seeing your change I'm struck by the duplication of these two contexts (running with list and with no subcommand). We (and by that I mean me...) should perhaps create an issue to combine these into a single context.

File: tests/chocolatey-tests/choco-template.Tests.ps1 [hunk 2]
```diff
@@ -163,5 +191,81 @@ Describe "choco <_>" -ForEach @(
...
+            $Output.Lines | Should -Contain "### Chocolatey Zip template"
+            $Output.Lines | Should -Contain "This adds a template for archive (zipped) packages."
+            $Output.Lines | Should -Contain "List of files:"
+            $Output.Lines | Should -Contain "$env:ChocolateyInstall\templates\zip\zip.nuspec"
+            $Output.Lines | Should -Contain "$env:ChocolateyInstall\templates\zip\ReadMe.md"
+            $Output.Lines | Should -Contain "$env:ChocolateyInstall\templates\zip\tools\chocolateybeforemodify.ps1"
+            $Output.Lines | Should -Contain "$env:ChocolateyInstall\templates\zip\tools\chocolateyinstall.ps1"
+            $Output.Lines | Should -Contain "$env:ChocolateyInstall\templates\zip\tools\LICENSE.txt"
+            $Output.Lines | Should -Contain "$env:ChocolateyInstall\templates\zip\tools\VERIFICATION.txt"
+        }
```
Comment (corbob) [inline tests/chocolatey-tests/choco-template.Tests.ps1:244]: These two it blocks are very close to the same. I wonder if we could streamline this with a single it block and a foreach inside 🤔 

Perhaps something like:

```powershell
It "Displays <Name> template information" -ForEach @(
    @{ Name = 'msi' ; Lines = @( <Insert lines here> )
    @{ Name = 'zip' ; Lines = @( <Insert lines here> )
) {
    Foreach ( $expectedLine in $Lines ) {
        $Output.Lines | Should -Contain $expectedLine
    }
}
```

**Review by corbob — CHANGES_REQUESTED** [review]
Just a couple of notes, nothing major.

Possibly the biggest thins I'm struggling with in this one is: can we possibly gate these tests without adding a whole bunch of skips... Or should we accept that many of these will possibly fail on 1.1.0 and just leave it at that?

File: tests/chocolatey-tests/choco-template.Tests.ps1 (hunk 1 above)
Comment (AdmiringWorm) [reply tests/chocolatey-tests/choco-template.Tests.ps1:44]: True, should probably look at a way to reduce that duplication. I don't think that should be something we should take into account right now though.

Comment (corbob) [reply tests/chocolatey-tests/choco-template.Tests.ps1:44]: I've opened a new issue to track this. Will update the issue with proper details later.

**Review by corbob — APPROVED** [review]
LGTM!

**Comment by corbob** [comment]
Thanks for getting this fixed up @AdmiringWorm!

**Pull request merged**